### PR TITLE
[container] add Git Repo volume mount support

### PR DESCRIPTION
**Pull request by yolo3301**
Add Git Repo volume mount support

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6010
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py [hunk 1]
```diff
@@ -65,7 +65,12 @@ def load_arguments(self, _):
...
         c.argument('azure_file_volume_account_name', help='The name of the storage account that contains the Azure File share')
         c.argument('azure_file_volume_account_key', help='The storage account access key used to access the Azure File share')
-        c.argument('azure_file_volume_mount_path', validator=validate_volume_mount_path, help='The path within the container where the volume should be mounted. Must not contain colon (:).')
+        c.argument('azure_file_volume_mount_path', validator=validate_volume_mount_path, help='The path within the container where the azure file volume should be mounted. Must not contain colon (:).')
+
+    with self.argument_context('container create', arg_group='Git Repo Volume') as c:
+        c.argument('gitrepo_repository', help='The URL of a git repository to be mounted as a volume')
+        c.argument('gitrepo_directory', validator=validate_gitrepo_directory, help="The target directory path in the git repository. Must not contain '..'.")
+        c.argument('gitrepo_mount_path', validator=validate_volume_mount_path, help='The path within the container where the git repo volume should be mounted. Must not contain colon (:).')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: A small detail but for `gitrepo_directory ` you use `''` around `..` for 'must not contain' but for `gitrepo_mount_path` you use `()` around `:`.
For consistency just use one or the other IMO.

File: src/command_modules/azure-cli-container/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.1.22
+++++++
+* Add Git Repo volume mount parameters '--gitrepo-repository' '--gitrepo-directory' and '--gitrepo-mount-path'
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-container/HISTORY.rst]: The names are kinda long no?

Comment (yolo3301) [reply src/command_modules/azure-cli-container/HISTORY.rst]: How about `--gitrepo-url` `--gitrepo-dir`? I also want to add `--gitrepo-revision` in another iteration.

Comment (derekbekoe) [reply src/command_modules/azure-cli-container/HISTORY.rst]: @yolo3301 yes like those better.

**Pull request merged**